tools/runners/run-all.js

```diff
@@ -125,7 +125,28 @@ class Runner {
     }
 
     var unblockAppRunner = self.appRunner.makeBeforeStartPromise();
-    self._startMongoAsync().then(unblockAppRunner);
+
+    function startMongo(tries = 3) {
+      self._startMongoAsync().then(
+        ok => unblockAppRunner(),
+        error => {
+          --tries;
+          const left = tries + (tries === 1 ? " try" : " tries");
+          Console.error(
+            `Error starting Mongo (${left} left): ${error.message}`
+          );
+
+          if (tries > 0) {
+            self.mongoRunner.stop();
+            setTimeout(startMongo, 1000);
```

Comment:
It looks like this line will keep defaulting tries to 3 instead of decrementing.

Would this work instead?

`setTimeout(() => startMongo(tries), 1000);`

Or could pull the `tries` declaration outside of `startMongo`.